tests/qubit_codes/test_surface_code.py

```diff
@@ -430,6 +429,90 @@ def test_boundary(self, surface_code):
+def rectangular_surface_code(request):
+    """A function to define a rectangular surface code for use in this
+    module."""
+    distance, ec, boundaries = request.param
+    return SurfaceCode(distance, ec, boundaries), request.param
+
+
+class TestRectangularSurfaceCode:
+    """Test the SurfaceCode class for non-cubic lattices."""
+
+    def test_init_rectangular(self, rectangular_surface_code):
+        """Check the proper initialization of SurfaceCode."""
+        # obtaining a rectangular surface code
+        rect_sc, param = rectangular_surface_code
+        distance, ec, boundaries = param
+
+        # assert that the surface code is initialized correctly
+        assert len(rect_sc.dims) == 3, "Dimensionality of the surface code is not 3 but {}".format(
+            len(rect_sc.dims)
+        )
+        for i, x in enumerate(rect_sc.dims):
+            assert isinstance(
+                x, (int, np.integer)
+            ), "Entry {} of distance {} is not an integer but {}".format(i, rect_sc.dims, type(x))
+            assert x == distance[i], "Entry {} of distance {} is not intended distance {}".format(
+                i, rect_sc.dims, distance[i]
+            )
+        assert rect_sc.ec
+        assert rect_sc.bound_str
+        assert list(rect_sc.boundaries)
+        for ec in rect_sc.ec:
+            attributes = [
+                "stabilizers",
+                "syndrome_inds",
+                "syndrome_coords",
+                "bound_points",
+                "stab_graph",
+            ]
+            for att in attributes:
+                getattr(rect_sc, ec + "_" + att)
+            for att in ["all_syndrome_inds", "all_syndrome_coords"]:
+                getattr(rect_sc, att)
+
+    def test_rectangular_stabilizers(self, rectangular_surface_code):
+        """Check whether all stabilizers were generated as expected."""
+        rect_sc, param = rectangular_surface_code
+
+        for ec in rect_sc.ec:
+            dx, dy, dz = rect_sc.dims
+            cubes = getattr(rect_sc, ec + "_stabilizers")
+            # Check that there are a correct number of stabilizer elements
+            # depending on the boundary conditions.
+            if rect_sc.bound_str.startswith("open"):
+                if len(rect_sc.ec) == 2 and ec == "dual":
+                    assert (
+                        len(cubes) == (dx - 1) * (dy - 1) * dz
+                    ), "Wrong number of cubes for open both or dual"
+                else:
+                    # I don't know what dimension is requires the -1
+                    assert (
+                        len(cubes) == dx * (dy - 1) * dz
+                    ), "1 Wrong number of cubes for open primal"
+                    # I don't know what dimension is requires the -1
+                    assert (
+                        len(cubes) == (dx - 1) * dy * dz
+                    ), "2 Wrong number of cubes for open primal"
```

Comment:
@sduquemesa this is the relevant part

Reply:
I'm trying to translate [this logic](https://github.com/XanaduAI/flamingpy/pull/61/files#diff-95be714889fac8d8a7943cf4664e911e1edb2b2ffb1c97d875833a2e248cb123R381-R387) to non-cubic codes: